forui/lib/src/widgets/text_field/text_field.dart

```diff
@@ -1062,6 +1187,359 @@ class FTextField extends StatefulWidget {
+      ..add(DiagnosticsProperty<FValueNotifier<bool>?>('visibilityController', visibilityController))
+      ..add(ObjectFlagProperty<FPasswordToggleBuilder?>.has('toggleBuilder', toggleBuilder));
+  }
+}
+
+class _PasswordFieldWithToggleState extends State<_PasswordFieldWithToggle> {
+  late FValueNotifier<bool> _visibilityController;
+  bool _shouldDisposeController = false;
+
+  @override
+  void initState() {
+    super.initState();
+    if (widget.visibilityController != null) {
+      _visibilityController = widget.visibilityController!;
+      _shouldDisposeController = false;
+    } else {
+      _visibilityController = FValueNotifier<bool>(false);
+      _shouldDisposeController = true;
+    }
+  }
+
+  @override
+  void didUpdateWidget(_PasswordFieldWithToggle oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (widget.visibilityController != oldWidget.visibilityController) {
+      if (_shouldDisposeController) {
+        _visibilityController.dispose();
+      }
+
+      if (widget.visibilityController != null) {
+        _visibilityController = widget.visibilityController!;
+        _shouldDisposeController = false;
+      } else {
+        _visibilityController = FValueNotifier<bool>(false);
+        _shouldDisposeController = true;
+      }
+    }
+  }
+
+  @override
+  void dispose() {
+    if (_shouldDisposeController) {
+      _visibilityController.dispose();
+    }
+    super.dispose();
+  }
+
+  Widget _buildToggleWidget(BuildContext context, FTextFieldStyle style, Set<WidgetState> states, bool isVisible) {
+    if (widget.toggleBuilder == FTextField._defaultToggleBuilder) {
+      // Use internal default implementation
+      final icon = isVisible ? FIcons.eyeOff : FIcons.eye;
+
+      return FButton.icon(
+        onPress: () => _visibilityController.value = !_visibilityController.value,
+        style: FButtonStyle.ghost(),
+        child: Icon(icon, semanticLabel: isVisible ? 'Hide password' : 'Show password'),
+      );
+    } else {
+      // Use custom toggle builder - user must manage controller themselves
+      // Note: This approach requires users to provide their own controller if using custom toggle
+      return widget.toggleBuilder!(context, style, states);
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) => ValueListenableBuilder<bool>(
+    valueListenable: _visibilityController,
+    builder: (context, isVisible, child) {
+      // Determine the suffix builder based on priority:
+      // 1. Custom suffixBuilder (highest priority - disables toggle)
+      // 2. toggleBuilder (if not null, shows toggle using internal controller for default, requires external controller for custom)
+      // 3. null (no suffix)
+      final FFieldIconBuilder<FTextFieldStyle>? effectiveSuffixBuilder =
+          widget.suffixBuilder ??
+          (widget.toggleBuilder != null
+              ? (context, style, states) => _buildToggleWidget(context, style, states, isVisible)
+              : null);
```

Comment:
I think we should either:
* Automatically wrap these in a row. 
* Change the API to accept either a suffix builder or toggle builder, not both.